src/blas/KokkosBlas1_iamax.hpp

```diff
@@ -90,10 +90,10 @@ typename XVector::size_type iamax (const XVector& x)
 /// Replace each entry in R with the (smallest) index of the element of the maximum magnitude of the
 /// corresponding entry in X.
 ///
-/// \tparam RMV 1-D or 2-D Kokkos::View specialization.
-/// \tparam XMV 1-D or 2-D Kokkos::View specialization.  It must have
-///   the same rank as RMV, and its entries must be assignable to
-///   those of RMV.
+/// \tparam RMV 0-D or 1-D Kokkos::View specialization.
+/// \tparam XMV 1-D or 2-D Kokkos::View specialization.
+///
+/// Special note for TPL cuBLAS:  RMV must be 0-D view and XMV must be 1-D view, and the index returned in RMV is 1-based since cuBLAS uses 1-based indexing for compatibility with Fortran
```

Comment:
The index needs to be the same, regardless of the implementation.  If that means subtracting 1 to convert from 1-based to 0-based indexing, then please do so.

Reply:
@mhoemmen Understand. I already did the subtraction when the cuBLAS functions return a value on host. 
But when the cuBLAS functions return directly the value on device memory, I do not know how to the subtraction efficiently. I do not want to call a kernel with only one thread for just doing the change for a single value so I leave the subtraction for user's kernel.
What would be the good way to do it? Can I just return the value to host and copy it to device if RMV is 0-D view?

Reply:
This is a kokkos-kernels design decision.  I would say, if the BLAS returns a 1-based index, then kokkos-kernels should return a 1-based index.  Kokkos users who want a 0-based index could easily implement their own, using `Kokkos::MaxLoc`.  I'm almost certain that a single kernel that kokkos-kernels provide would be faster than calling cuBLAS and then invoking another kernel just to decrement a single value.

Reply:
@srajama1 What is your opinion? Should we use 0-based index or 1-based index?

Reply:
There is no good answer for this in the decades we have dealt with it. I recommend to do whatever is most efficient for our users, we can jump through hoops to help them.

Reply:
@srajama1 @mhoemmen For now, I think I would use 0-based index for all cases, except when TPL cuBLAS iamax is used and returns result to a 0-D view on device memory. I would leave users doing the 1-based to 0-based conversion in their kernels if they need to. I added note to specify this case in the ```KokkosBlas1_iamax.hpp```

However, I am open to changing to 0-based indexing for cuBLAS or using 1-based indexing for all cases. Please let me know.

Reply:
Can we add a note in the wiki, explaining this ?

Reply:
Thanks, @srajama1  I will take care of the wiki.
@mhoemmen Could you please approve this so that I can run spotcheck and get it merged?